src/testUtils/testUtilsThree.tsx

```diff
@@ -0,0 +1,71 @@
+import { beforeEach, afterEach } from 'vitest'
+import { create } from '@react-three/test-renderer'
+import * as React from 'react'
+
+// https://github.com/testing-library/react-testing-library/blob/main/src/act-compat.js
+function getGlobalThis() {
+  /* istanbul ignore else */
+  if (typeof globalThis !== 'undefined') {
+    return globalThis
+  }
+  /* istanbul ignore next */
+  if (typeof self !== 'undefined') {
+    return self
+  }
+  /* istanbul ignore next */
+  if (typeof window !== 'undefined') {
+    return window
+  }
+  /* istanbul ignore next */
+  if (typeof global !== 'undefined') {
+    return global
+  }
+  /* istanbul ignore next */
+  throw new Error('unable to locate global object')
+}
```

Comment:
In order for `act` to work, the scheduler has to be mocked to `scheduler/unstable_mock`. Perhaps `IS_REACT_ACT_ENVIRONMENT` etc. can be moved to a setup file?

Reply:
> Perhaps IS_REACT_ACT_ENVIRONMENT etc. can be moved to a setup file?

I honestly don't know how it's supposed to be handled. In react testing library they wrap the act itself, set `IS_REACT_ACT_ENVIRONMENT` to true before executing callback in act and set `IS_REACT_ACT_ENVIRONMENT` back to previous value when act is completed. I'm not sure why they do it, but this is the reason why I thought that setting it to true across all test via setupFile.

Reply:
So just move the code you suggested to a setupFile?

Reply:
seems to work just fine